x-pack/legacy/plugins/siem/server/lib/detection_engine/types.ts

```diff
@@ -30,6 +32,7 @@ export interface ThreatParams {
 export type RuleType = 'query' | 'saved_query' | 'machine_learning';
 
 export interface RuleAlertParams {
+  actions: AlertAction[] | undefined;
```

Comment:
Out of curiosity: is `| undefined` for backwards compatibility? Is there a current situation where one can create a rule with `undefined` actions?